https://pagure.io/fedora-infrastructure/issue/8560

--- /usr/share/copr/backend/msgbus.py	2020-01-15 08:23:09.000000000 +0000
+++ /usr/share/copr/backend/msgbus.py	2020-01-20 09:53:36.100341122 +0000
@@ -181,6 +181,7 @@
         for attempt in range(1, self.opts.bus_publish_retries + 1):
             try:
                 self._send_message(message)
+                break
             except Exception: # pylint: disable=W0703
                 # We don't want to halt the worker because of messaging.
                 self.log.exception("Attempt %s to publish a message failed", attempt)
